doc/help/elektra-highlevel-gen.md

```diff
@@ -0,0 +1,413 @@
+# elektra-highlevel-gen(7) -- High-level API code-generation advanced features
+
+This document focuses on the advanced features of the high-level API code-generator template. For the basic features see
+[`kdb-gen-highlevel(1)](kdb-gen-highlevel.md)
+
+## Configuration Options
+
+The parameters that are relevant to the concepts described here are (for the rest see [`kdb-gen-highlevel(1)](kdb-gen-highlevel.md)):
+
+- `enumConv`: allowed values: `strcmp`, `trie`, `default` (default)
+- `specLocation`: allowed values: `embedded` (default), `external`
+- `defaultsHandling`: allowed values: `embedded` (default), `speconly`
+- `specValidation`: allowed values: `none` (default), `minimal`
+
+The `enumConv` option is described [below](#enums).
+
+With `specLocation` and `defaultsHandling` you can configure what data is embedded into your application's binary. The default option for
+both is `embedded`. This embeds the full specification into your application. It should then be mounted into the KDB via `specload` and will
+also be used as the `defaults` keyset passed to the high-level API's `elektraOpen`. The idea behind this is that the compilation still
+results in a single file.
+
+If you don't care that additional files will be required to run your application, you can switch to `specLocation=external`. With this option
+the code-generator creates an additional output file, which contains the processed specification. The files is in `quickdump` format, so it
+can be mounted directly via `quickdump`. If embedding the specification into your application is not something you want to do (e.g. because
+of the size increase), but you still want to use `specload`, you can do that too. Just use a configuration like
+`app=/usr/bin/cat args=#0 args/#0="path-to-spec-output-file"`.
+
+Using the combination `specLocation=external defaultsHandling=embedded`, still embeds a smaller keyset only containing the default values
+into your application. If that is still too much for you, may use `defaultsHandling=embedded`. This way defaults are not passed to
```

Comment:
defaultsHandling=embedded already used before?

Reply:
Copy-paste error, it should obviously be `defaultsHandling=speconly` (the other setting for `defaultsHandling`)

Reply:
Was not clear to me. You need to be very careful in these explanations as this configuration is unfortunately very complicated.